### DISCUSSION TITLE
Non-matching closing bracket in a block of code

### DISCUSSION THREAD

**Discussion by adamantinum** [helix-editor/helix#8574] [General]
I don't know if it's the expected behaviour, but if I'm writing a block of code delimited by parentesis I would expect, starting a new line and closing the bracket, to match the closing bracket created by auto-pairing.

E.g.:
```c++
{
    |     // here I start a new line and type '}'
}
``` 

I get

```c++
{
    
    }|
}
```

**Comment by pascalkuthe** [comment]
This is working as intended. I don't really see the issue here. Do you mean that you want the bracket to be automaticlly dedented? That would be #6044

**Comment by adamantinum** [comment]
It should also use the previously autoinserted closing bracket, instead of adding an exceeding one

**Comment by pascalkuthe** [comment]
Ah so you mean how you can type `}`  at `{|}` to move the cursor to the right `{}|`.

Yeah I don't think we will implement this. That would mean we would skip arbitrary whitespace (including newlines).

`{| }` also inserts an extra bracket on typing `}`.

Changing this would leave to a lot of annoying false positives.

**Comment by rhogenson** [comment]
I'm also interested in this feature, it would be nice not to have to leave insert mode to skip over the auto-inserted closing brace. I think it could minimize false positives by only skipping over the closing brace if it was inserted automatically as an auto-pair
